mlflow/utils/name_utils.py

```diff
@@ -338,3 +368,264 @@ def _generate_random_name(sep="-", integer_scale=3, max_length=20):
     "youthful",
     "zealous",
 ]
+
+
+# Unique dataset names are generated from dataset digests by mapping the digest to an integer
+# and applying the modulo operation with a prime divisor to obtain an index. A prime divisor
+# is used to reduce the likelihood of collisions. This index is then mapped to a predicate
+# (e.g. "unique") and an integer in the range [0, 100000) (e.g. 9054). Finally, the dataset name
+# is constructed as <predicate>-data-<integer> (e.g. unique-data-9054)
+_GENERATOR_DATASET_DIGEST_MODULO_DIVISOR = 24999983
+_GENERATOR_DATASET_INTEGER_UPPER_BOUND = 10**5
+_GENERATOR_DATASET_PREDICATES = [
```

Comment:
Introduce new predicates for datasets to distinguish them from runs. Run names also use some predicates that sound okay with animals, e.g. "youthful-unicorn-7" but sound a bit more odd with data "youthful-data-324".